app/controllers/volunteers_controller.rb

```diff
@@ -305,4 +305,17 @@ def home
 
     render :home
   end
+
+  private
+
+  def my_admin_volunteers(admin_region_ids)
+    Volunteer.all.collect do |volunteer|
```

Comment:
geebuz no wonder this view can timeout... this loads / iterates through ~2500 volunteers in production

Reply:
Yes, perhaps that needs a tweak.

Reply:
Yeah feels like we could select Volunteer based on the region ids we already have. 

from volunteer.rb
`query.joins(:regions).where(regions: { id: region_ids })`

So maybe `Volunteer.joins(:regions).where(regions: {id: admin_region_ids}}`?

Reply:
We could also do `Volunteer.find_each.collect do|volunteer|` no?

Reply:
We could also do `Volunteer.find_each.collect do|volunteer|` no?

Reply:
That'd be in memory,  too. yes?

Reply:
from http://guides.rubyonrails.org/active_record_querying.html

"The find_each and find_in_batches methods are intended for use in the batch processing of a large number of records that wouldn't fit in memory all at once. If you just need to loop over a thousand records the regular find methods are the preferred option."

Reply:
By using region_ids instead of just regions, we are forcing ActiveRecord to run a query and return an Array, and then we use that Array in the next query. I think it may be more efficient to use regions and then let ActiveRecord decide when to make the query, as this would (I think) result in just one query. Also, I think we want to add distinct to the query to avoid duplicate records:

`@volunteers = Volunteer.joins(:regions).where(regions: {id: current_volunteer.regions}).distinct`

This should result in a single query looking something like:

`SELECT DISTINCT "volunteers".* FROM "volunteers" INNER JOIN "assignments" ON "assignments"."volunteer_id" = "volunteers"."id" INNER JOIN "regions" ON "regions"."id" = "assignments"."regions_id" WHERE "regions"."id" IN (SELECT "regions"."id" FROM "regions" INNER JOIN "assignments" ON "regions"."id" = "assignments"."region_id" WHERE "assignments"."volunteer_id" = $1)`

And it should return an ActiveRecord Relation, allowing us to further process before sending. But I may be missing something, as I haven't been close to this project. What do you think?